packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2594,22 +2593,46 @@ class RenderPointerListener extends RenderProxyBoxWithHitTestBehavior {
     }
   }
 
+  void _handleMouseTrackerChanged() {
+    if (attached)
+      markNeedsPaint();
+  }
+
   @override
   void attach(PipelineOwner owner) {
     super.attach(owner);
     // Add a listener to listen for changes in mouseIsConnected.
-    RendererBinding.instance.mouseTracker.addListener(_updateAnnotations);
+    RendererBinding.instance.mouseTracker.addListener(_handleMouseTrackerChanged);
+    postActivate();
+  }
+
+  /// Attaches the annotation for this render object, if any.
+  ///
+  /// This is called by [attach], and the [Listener]'s element to tell the
```

Comment:
uber-nit: Any chance you can simplify this sentence a little? Maybe split it up in multiple sentence. It's hard to read going over four lines and with the two "so that" in there..

Reply:
Done.